mocks/mocks.go

```diff
@@ -238,6 +239,27 @@ func (sa *StorageAuthority) GetAuthorization(_ context.Context, id string) (core
 		return core.Authorization{}, fmt.Errorf("Unspecified database error")
 	}
 
+	prob := &probs.ProblemDetails{
+		Type:       "things:are:whack",
+		Detail:     "whack attack",
+		HTTPStatus: 555,
+	}
+	exp := sa.clk.Now().AddDate(100, 0, 0)
+	authz.Expires = &exp
+	// "oldNS" returns an authz with a failed challenge that has the problem type
+	// statically prefixed by the V1ErrorNS
+	if id == "oldNS" {
```

Comment:
We follow this pattern in some of our mocks - have a unique string identifier that elicits one behavior vs another. I worry that it creates spooky action at a distance that may make debugging or modifying our tests hard. There's another approach, where we create a type for a specific test case that embeds `mocks.StorageAuthority` and overrides just a single method, e.g. https://github.com/letsencrypt/boulder/blob/master/ra/ra_test.go#L687. I think that might be a suitable improvement here.

Reply:
Sounds good. Implemented in b44adcc. I put the new mock in `mocks/` instead of along side the tests (unlike the [RA example you linked](https://github.com/letsencrypt/boulder/blob/master/ra/ra_test.go#L687)) because otherwise it will have to be duplicated in both the `wfe` and `wfe2` package.